### PR TITLE
DOCS: Added more detail to titleTemplate.

### LINKED ISSUE #4984: Cannot set `titleTemplate` as function
### Environment

------------------------------
- Operating System: `Darwin`
- Node Version:     `v14.19.2`
- Nuxt Version:     `3.0.0-rc.3`
- Package Manager:  `yarn@1.22.18`
- Builder:          `vite`
- User Config:      `modules`, `target`, `app`, `content`, `css`, `build`
- Runtime Modules:  `@nuxt/content-edge@2.0.0-27541979.40705ae`
- Build Modules:    `-`
------------------------------

### Reproduction

N/A

### Describe the bug

Cannot set the `titleTemplate` as a function (used for default titles when non existent).

Example:
```js
import { defineNuxtConfig } from 'nuxt'

// https://v3.nuxtjs.org/api/configuration/nuxt.config
export default defineNuxtConfig({
    app: {
        head: {
            titleTemplate: (titleChunk) => {
                return titleChunk ? `${titleChunk} - Site Title` : 'Site Title';
            },
        }
    }
})
```

### Additional context

As far as I know, Nuxt 3 uses vue-meta, which the docs show it supports: https://vue-meta.nuxtjs.org/api/#titletemplate

### Logs

_No response_

### PR DESCRIPTION
**Pull request by heychazza**
### 🔗 Linked issue

#4984

### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

This change adds more clarity to the `titleTemplate` option, as currently in Nuxt 3 this doesn't work as intended. This resolves #4984.

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: docs/content/2.guide/2.features/4.head-management.md [hunk 1]
```diff
@@ -31,6 +29,32 @@ useHead({
...
+  useHead({
+    titleTemplate: (titleChunk) => {
+      return titleChunk ? `${titleChunk} - Site Title` : 'Site Title';
+    },
+  });
+});
+```
+
+Then you can set a title as you usually would:
+```js
```
Comment (manniL) [inline docs/content/2.guide/2.features/4.head-management.md:48]:
```suggestion
```vue
```

**Comment by heychazza** [comment]
Preview: https://deploy-preview-5057--nuxt3-docs.netlify.app/guide/features/head-management#title-templates

File: docs/content/2.guide/2.features/4.head-management.md [hunk 2]
```diff
@@ -31,6 +29,32 @@ useHead({
...
+    titleTemplate: (titleChunk) => {
+      return titleChunk ? `${titleChunk} - Site Title` : 'Site Title';
+    },
+  });
+});
+```
+
+Then you can set a title as you usually would:
+```vue
+// pages/my-page.vue
```
Comment (DamianGlowala) [inline docs/content/2.guide/2.features/4.head-management.md:49]:
```suggestion
```vue [/pages/my-page.vue]
```

File: docs/content/2.guide/2.features/4.head-management.md [hunk 3]
```diff
@@ -31,6 +29,32 @@ useHead({
...
+<script setup>
+  useHead({
+    title: 'My Page'
+  });
+</script>
+```
+This will now appear as 'My Page - Site Title' in the browser tab, you can also pass `null` to default to the site title.
```
Comment (DamianGlowala) [inline docs/content/2.guide/2.features/4.head-management.md:56]:
```suggestion
This will now appear as 'My Page - Site Title' in the browser tab. You can also pass `null` to default to the site title.
```

**Review by DamianGlowala — CHANGES_REQUESTED** [review]

File: docs/content/2.guide/2.features/4.head-management.md [hunk 4]
```diff
@@ -31,6 +29,31 @@ useHead({
 ::ReadMore{link="/api/composables/use-head"}
 ::
 
+## Title Templates
+In previous versions of Nuxt, you could use the `titleTemplate` option to dynamically customize the title of your site. However, this is no longer supported in Nuxt 3, and can instead be done via a plugin.
```
Comment (danielroe) [inline docs/content/2.guide/2.features/4.head-management.md:33]:
```suggestion
You can use the `titleTemplate` option to provide a dynamic template for customizing the title of your site, for example, by adding the name of your site to the title of every page.
```

File: docs/content/2.guide/2.features/4.head-management.md [hunk 5]
```diff
@@ -31,6 +29,31 @@ useHead({
...
+
+Create a `title.js` plugin within your `plugins` folder, with the following code:
```
Comment (danielroe) [inline docs/content/2.guide/2.features/4.head-management.md:35]:
```suggestion
The `titleTemplate` can either be a string, where `%s` is replaced with the title, or a function. If you want to use a function (for full control), then this cannot be set in your `nuxt.config`, and it is recommended instead to set it within your `app.vue` file, where it will apply to all pages on your site:
```

File: docs/content/2.guide/2.features/4.head-management.md [hunk 6]
```diff
@@ -31,6 +29,31 @@ useHead({
...
+export default defineNuxtPlugin(() => {
+  useHead({
+    titleTemplate: (titleChunk) => {
+      return titleChunk ? `${titleChunk} - Site Title` : 'Site Title';
+    },
+  });
+});
+```
+
+Then you can set a title as you usually would:
```
Comment (danielroe) [inline docs/content/2.guide/2.features/4.head-management.md:47]:
```suggestion
```

File: docs/content/2.guide/2.features/4.head-management.md [hunk 7]
```diff
@@ -31,6 +29,31 @@ useHead({
...
+```vue [/pages/my-page.vue]
+<script setup>
+  useHead({
+    title: 'My Page'
```
Comment (danielroe) [inline docs/content/2.guide/2.features/4.head-management.md:51]:
```suggestion
    titleTemplate: (titleChunk) => {
      return titleChunk ? `${titleChunk} - Site Title` : 'Site Title';
    },
```

File: docs/content/2.guide/2.features/4.head-management.md [hunk 8]
```diff
@@ -31,6 +29,31 @@ useHead({
...
+  });
+</script>
+```
+This will now appear as 'My Page - Site Title' in the browser tab. You can also pass `null` to default to the site title.
```
Comment (danielroe) [inline docs/content/2.guide/2.features/4.head-management.md:55]:
```suggestion

Now, if you set the title to `My Page` with `useHead` on another page of your site, the title would appear as 'My Page - Site Title' in the browser tab. You can also pass `null` to default to the site title.
```

File: docs/content/2.guide/2.features/4.head-management.md (hunk 7 above)
Comment (danielroe) [reply docs/content/2.guide/2.features/4.head-management.md:51]: the function approach _should_ work everywhere but in `nuxt.config`.

Comment (heychazza) [reply docs/content/2.guide/2.features/4.head-management.md:51]: Even with the following example in `app.vue`:
```vue
<script setup>
useHead({
  titleTemplate: (titleChunk) => {
    return titleChunk ? `${titleChunk} - Site` : 'Site: Default Text';
  },
});
</script>

<template>
  <div>
    <NuxtLayout>
      <NuxtPage/>
    </NuxtLayout>
  </div>
</template>
```

If you don't pass a `useHead` to your page, it won't default it.

**Comment by pi0** [comment]
Thanks for working on this amazing PR @heychazza @manniL @danielroe 💚 

I believe before adding docs and instead of advising to use a Nuxt plugin, we can improve experience by suggesting to either use new `app.vue` which is meant to be entry of global setup such as layout and styles, it can also be used to declare titleTemplate without an additional file.

Additionally, we could introduce `app/router.options` support for runtime function.

**Comment by danielroe** [comment]
@pi0 Agreed. I think my pending review comments should exactly implement that 👍

**Comment by heychazza** [comment]
The only issue I've personally found, but this would be a separate issue :-

Setting it within `app.vue`, if you don't specify a title within a page, it'll show up as null.

I'll be back on PC in ~2 hours so I could make a reproduction repo if needed.

TL;DR is if no title is specified in pages, it doesn't use the default from the `titleTemplate`.

**Comment by danielroe** [comment]
@heychazza I have [submitted a PR](https://github.com/nuxt/framework/pull/5064) to fix that 😉

**Comment by heychazza** [comment]
> @heychazza I have [submitted a PR](https://github.com/nuxt/framework/pull/5064) to fix that 😉 

Smh Daniel, beating me to it 😂

Thanks though, haha

**Comment by heychazza** [comment]
Oh eek, I tried to do another PR, so I reset my main repo, apologies, will fix that and re-send it.
